src/messaging/ExchangeContext.cpp

```diff
@@ -87,24 +87,30 @@ void ExchangeContext::SetResponseTimeout(Timeout timeout)
 #if CONFIG_DEVICE_LAYER && CHIP_DEVICE_CONFIG_ENABLE_SED
 void ExchangeContext::UpdateSEDPollingMode()
 {
-    SessionHandle sessionHandle              = GetSessionHandle();
-    Transport::Session::SessionType sessType = sessionHandle->GetSessionType();
+    Transport::PeerAddress address;
 
-    // During PASE session, which happen on BLE, the session is kUnauthenticated
-    // So AsSecureSession() ends up faulting the system
-    if (sessType != Transport::Session::SessionType::kUnauthenticated)
+    switch (GetSessionHandle()->GetSessionType())
     {
-        if (sessionHandle->AsSecureSession()->GetPeerAddress().GetTransportType() != Transport::Type::kBle)
-        {
-            if (!IsResponseExpected() && !IsSendExpected() && (mExchangeMgr->GetNumActiveExchanges() == 1))
-            {
-                chip::DeviceLayer::ConnectivityMgr().RequestSEDFastPollingMode(false);
-            }
-            else
-            {
-                chip::DeviceLayer::ConnectivityMgr().RequestSEDFastPollingMode(true);
-            }
-        }
+    case Transport::Session::SessionType::kSecure:
+        address = GetSessionHandle()->AsSecureSession()->GetPeerAddress();
+        break;
+    case Transport::Session::SessionType::kUnauthenticated:
+        address = GetSessionHandle()->AsUnauthenticatedSession()->GetPeerAddress();
+        break;
+    default:
+        return;
+    }
+
+    VerifyOrReturn(address.GetTransportType() != Transport::Type::kBle);
+    UpdateSEDPollingMode(IsResponseExpected() || IsSendExpected() || IsMessageNotAcked() || IsAckPending());
```

Comment:
If IsAckPending() but not either IsResponseExpected() or IsSendExpected(), then we're about to send the ack as soon as we return from this function (due to closing the exchange).   Is the idea that we need to stay in fast-poll mode until we have sent it?

And if we do have an ack pending and we're about to close the exchange, should we then exit fast-poll mode after we send the ack?  Or are we just hoping that our destructor will happen soon after?

Reply:
I think if the ACK is delivered successfully then there's no need to stay in the fast-polling mode as it's only useful for receiving messages. But I wonder what if the ACK is lost. Do we then release an exchange and create another one when the last message is retransmitted? I need to check that, but I think you're right and I should remove `IsAckPending` from the condition.

Reply:
I've decided to remove `IsAckPending()` as we don't need fast-polling for sending the ACK.

Reply:
> Do we then release an exchange and create another one when the last message is retransmitted? 

Yes.